projects/core/src/lib/filter/helpers/cql-spatial-filter.helper.ts

```diff
@@ -0,0 +1,68 @@
+import { SpatialFilterModel } from '../models/spatial-filter.model';
+
+export class CqlSpatialFilterHelper {
+
+  public static convertSpatialFilterToQuery(filter: SpatialFilterModel, layerId: number): string | null {
+    if (filter.geometries.length === 0 || filter.geometryColumns.length === 0) {
+      return null;
+    }
+
+    const { baseGeometries, circles } = this.categorizeGeometries(filter.geometries.map(g => g.geometry));
+    const filterGeometries: string[] = [];
+    if (baseGeometries.length > 0) {
+      const geometryCollection = 'GEOMETRYCOLLECTION(' + baseGeometries.join(',') + ')';
```

Comment:
We can add case for `length === 1` for the geometry not wrapped in a geometry collection.